dragons-nft-tests/src/cep47.rs

```diff
@@ -112,154 +140,161 @@ impl CasperCEP47Contract {
+        token_metas: &Vec<Meta>,
+        sender: &AccountHash,
+    ) {
         self.call(
             sender,
             "mint_many",
             runtime_args! {
-                "recipient" => recipient,
-                "token_metas" => token_metas
+                "recipient" => Key::from(*recipient),
+                "token_ids" => token_ids.map(|ids| ids.clone()),
+                "token_metas" => token_metas.clone(),
             },
         );
     }
 
-    pub fn burn_many(&mut self, owner: PublicKey, token_ids: Vec<TokenId>, sender: Sender) {
+    pub fn burn_many(
+        &mut self,
+        owner: &AccountHash,
+        token_ids: &Vec<TokenId>,
+        sender: &AccountHash,
+    ) {
         self.call(
             sender,
             "burn_many",
             runtime_args! {
-                "owner" => owner,
-                "token_ids" => token_ids
+                "owner" => Key::from(*owner),
+                "token_ids" => token_ids.clone()
             },
         );
     }
 
-    pub fn burn_one(&mut self, owner: PublicKey, token_id: TokenId, sender: Sender) {
+    pub fn burn_one(&mut self, owner: &AccountHash, token_id: &TokenId, sender: &AccountHash) {
         self.call(
             sender,
             "burn_one",
             runtime_args! {
-                "owner" => owner,
-                "token_id" => token_id
+                "owner" => Key::from(*owner),
+                "token_id" => token_id.clone()
             },
         );
     }
 
     pub fn transfer_token(
         &mut self,
-        owner: PublicKey,
-        recipient: PublicKey,
-        token_id: TokenId,
-        sender: Sender,
+        recipient: &AccountHash,
+        token_id: &TokenId,
+        sender: &AccountHash,
     ) {
         self.call(
             sender,
             "transfer_token",
             runtime_args! {
-                "sender" => owner,
-                "recipient" => recipient,
-                "token_id" => token_id
+                "recipient" => Key::from(*recipient),
+                "token_id" => token_id.clone()
             },
         );
     }
 
     pub fn transfer_many_tokens(
         &mut self,
-        owner: PublicKey,
-        recipient: PublicKey,
-        token_ids: Vec<TokenId>,
-        sender: Sender,
+        recipient: &AccountHash,
+        token_ids: &Vec<TokenId>,
+        sender: &AccountHash,
     ) {
         self.call(
             sender,
             "transfer_many_tokens",
             runtime_args! {
-                "sender" => owner,
-                "recipient" => recipient,
-                "token_ids" => token_ids
+                "recipient" => Key::from(*recipient),
+                "token_ids" => token_ids.clone()
             },
         );
     }
 
-    pub fn transfer_all_tokens(&mut self, owner: PublicKey, recipient: PublicKey, sender: Sender) {
+    pub fn transfer_all_tokens(&mut self, recipient: &AccountHash, sender: &AccountHash) {
```

Comment:
This does not account for the possible "recipient is contract case"

Reply:
Let's leave it for now, but please change that in the PR with tests for smart contracts transfering tokens.